slack-plugin/src/main/kotlin/slack/gradle/lint/LintTasks.kt

```diff
@@ -50,26 +66,178 @@ internal object LintTasks {
+                slackProperties,
+                affectedProjects,
+                onProjectSkipped
+              )
+            }
+          }
+          // Only configure non-multiplatform Java projects via JavaPlugin. Multiplatform
+          // projects targeting Java (e.g. `jvm { withJava() }`) are configured via
+          // KotlinBasePlugin.
+          is JavaPlugin ->
+            if (project.multiplatformExtension == null) {
+              project.configureNonAndroidProjectForLint(
+                slackProperties,
+                affectedProjects,
+                onProjectSkipped
+              )
+            }
+          // Only configure non-Android multiplatform projects via KotlinBasePlugin.
+          // Multiplatform projects targeting Android (e.g. `id("com.android.library")`) are
+          // configured via AppPlugin or LibraryPlugin.
+          is KotlinBasePlugin ->
+            if (
+              project.multiplatformExtension != null &&
+                !project.plugins.hasPlugin(AppPlugin::class.java) &&
+                !project.plugins.hasPlugin(LibraryPlugin::class.java)
+            ) {
+              project.configureNonAndroidProjectForLint(
+                slackProperties,
+                affectedProjects,
+                onProjectSkipped
+              )
+            }
+        }
+      }
+    project.plugins.all(projectAllAction)
+  }
+
+  /** Android Lint configuration entry point for Android projects. */
+  private fun Project.configureAndroidProjectForLint(
+    slackProperties: SlackProperties,
+    affectedProjects: Set<String>?,
+    onProjectSkipped: (String, String) -> Unit,
+  ) =
+    androidExtension.finalizeDsl { extension ->
+      project.log("Applying ciLint to Android project")
+
+      project.log(
+        "Configuring android lint tasks. isApp=${extension is ApplicationAndroidComponentsExtension}"
+      )
+
+      project.log("Creating ciLint task")
+      val ciLintTask =
+        project.tasks.register(CI_LINT_TASK_NAME) { group = LifecycleBasePlugin.VERIFICATION_GROUP }
+
+      val ciLintVariants = slackProperties.ciLintVariants
+      if (ciLintVariants != null) {
+        ciLintTask.configure {
+          // Even if the task isn't created yet, we can do this by name alone and it will resolve at
+          // task configuration time.
+          ciLintVariants.splitToSequence(',').forEach { variant ->
+            logger.debug("Using variant $variant for ciLint task")
+            val lintTaskName = "lint${variant.capitalizeUS()}"
+            dependsOn(lintTaskName)
+          }
+        }
+      } else {
+        androidExtension.onVariants { variant ->
+          val lintTaskName = "lint${variant.name.capitalizeUS()}"
+          project.log("Adding $lintTaskName to ciLint task")
+          ciLintTask.configure {
+            // Even if the task isn't created yet, we can do this by name alone and it will resolve
+            // at
+            // task configuration time.
```

Comment:
weird splitting of comment across multiple lines

Reply:
ktfmt doin things

Reply:
[197a830](https://github.com/slackhq/slack-gradle-plugin/pull/612/commits/197a83093b06c9998823109712c340a17312ff80)